.github/workflows/ci.yml

```diff
@@ -1,59 +1,29 @@
 name: CI
-on:
-  pull_request:
-    branches: [ main ]
-
-# env: # testing if we need this..
-#   GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+on:
+  workflow_call:
+    inputs:
+      python-version:
+        description: Python version to use in the CI
+        required: false
+        type: string
+        default: '["3.10", "3.11", "3.12"]'
```

Comment:
I think it would be cleaner to have the job matrix in the calling workflow, instead of specifying multiple python versions here.